sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1761,6 +1761,8 @@ class Analyzer(
       // Only a few unary nodes (Project/Filter/Aggregate) can contain subqueries.
       case q: UnaryNode if q.childrenResolved =>
         resolveSubQueries(q, q.children)
+      case d: DeleteFromTable if d.childrenResolved =>
```

Comment:
isn't `DeleteFromTable` a unary node?

Reply:
Thank you @cloud-fan. Currently it is a `Command`. Maybe your suggestion is right, since `Command` is usually for DDLs and we need not `DeleteFromTable` be executed eagerly. Will update it.

Reply:
However, some similar operators like `OverwriteByExpression` is actually a `Command`. Should we consider this seriously?

Reply:
`Command` is a trait, `DeleteFromTable` can be both `UnaryNode` and `Command`.

Reply:
`UnaryNode` has a final `children` while there's a default override version of `children` in `Command`, which is a problem. A `Command` is designed for operators that has side-effects, so they can be executed eagerly. In this sense, I'd prefer to keep `DeleteFromTable` a `Command`. What do you think?

Reply:
makes sense

Reply:
I agree that `DeleteFromTable` should be a command that is executed eagerly because it has side-effects.